### ISSUE TITLE
Computing hallucinations for impossible trips

### ISSUE THREAD

**Issue by Lerri-Cofannos** [XavB64/lowtrip#57]
Issue reported by message

## Description:
In the "comparisons with other modes of transport," sometimes the algorithm attempts to produce a value that may simply not exist. For example, a journey obscure enough that there might not be a train route: Kiruna (Sweden) to Reine (Norway). Yet, the map produces a value for the train (and even a line that goes completely off track). It is still needed to verify if this also happens with airports. I believe there are few airports in the Lofoten area of Norway, where this could happen.

## Steps to Reproduce:

Select Kiruna (Sweden) as the origin and Reine (Norway) as the destination.
Observe the value and route suggested for the train.
Expected Behavior:
The algorithm should recognise when there is no viable train route and provide appropriate feedback or alternative suggestions.

**Comment by XavB64** [comment]
For now, this tool is not meant to be used as a "low carbon holiday planner". For this purpose, we recommend to look at https://mollow.eu for instance :)

This is because the paths geometries (for trains and buses) are just returned from the shortest path computed on the OSM network. There is no guaranty that the railway used is operated by any travel company.
Similarly, for planes or ferry a geometry is proposed all the time (even if it does not exist) if the user queries it. 

If you are interested, we can try to use APIs from travel companies to improve that, but to this day I have found none that are free of charge.

With that being said, the example you provided is interesting because Reine is on an island and the OSM network has a "tolerance" to the distance to the closest railway. This is the reason why a path is still returned. 

We've noticed that during the development and I therefore implemented a validation function which filters paths that are too far away from the queried location. Right now the parameter is at 100km (I recognize it is still pretty large indeed), but we can think of decreasing its value ?
